app/proguard-rules.pro

```diff
@@ -0,0 +1,31 @@
+-dontwarn
+-ignorewarnings
+
+# RetroFit
+-dontwarn retrofit.**
+-keep class retrofit.** { *; }
+-keepattributes Signature
+-keepattributes Exceptions
+
+# Glide
+-keep public class * implements com.bumptech.glide.module.GlideModule
```

Comment:
What exactly is this file doing? I've never actually had to use a proguard file for anything.

Reply:
ProGuard obfuscates and optimizes release binaries for security and performance reasons. As a part of that, it renames classes and members to auto-generated names like a.b.c.d, and also removes unused ones. This, however, causes issues when using reflection for e.g. serialization (like the API client does) and self-registering modules (like Glide does). ProGuard files like the newly created ones configure it to not obfuscate and/or remove some of the classes - most of the content here was copied from the Phonograph App itself.
Since you seem to have been using the debug build only for now, there were no issues - but on the release build, `minifyEnabled` is set, which executes ProGuard, and without the ProGuard config, the code breaks completely. Thus, I added one.